src/render/tutorials/graphql-usage/2_0.html.md

```diff
@@ -0,0 +1,777 @@
+```
+
+It's a standard Knot.x handler factory. It returns a handler that creates a graphQL object, uses it to create a Vert.x `GraphQLHandler` and lets it handle the request.
+
+`GraphQL` object is created the standard way. See [graphql-java.com](https://www.graphql-java.com/) for more explanation on Java GraphQL specific issues.
+
+```java
+private static GraphQL setupGraphQL(Vertx vertx, JsonObject config, RoutingContext routingContext) {
+    // Schema file path will be read from knot.x configuration
+    Reader schema = loadResource(config.getString("schema")); 
+
+    SchemaParser schemaParser = new SchemaParser();
+    TypeDefinitionRegistry typeDefinitionRegistry = schemaParser.parse(schema);
+
+    RuntimeWiring runtimeWiring = newRuntimeWiring()
+        .type("QueryType", builder -> builder
+            // Here we can assign data fetchers to operations from schema
+            // We'll implement them in Step 2. Normaly it'd look something like:
+            .dataFetcher("books", new BooksFetcher())
+            .dataFetcher("book", new SingleBookFetcher())
+        )
+        .build();
+    SchemaGenerator schemaGenerator = new SchemaGenerator();
+    GraphQLSchema graphQLSchema = schemaGenerator
+        .makeExecutableSchema(typeDefinitionRegistry, runtimeWiring);
+
+    return GraphQL.newGraphQL(graphQLSchema).build();
+  }
+
+private static Reader loadResource(String path) {
+    return new InputStreamReader(GraphQLHandlerFactory.class.getResourceAsStream("/" + path));
+}
+```
+
+Rememeber about updating services in META-INF. With a handler factory you can now configure a graphql endpoint. Add this path to `openapi.yml`:
+
+```yml
+/api/graphql:
+    post:
+        operationId: get-books-gql
+        responses:
+            default:
+                description: Books Graphql API
+```
+
+Note that it needs to be POST because GraphQL reads queries from the requests' bodies. Now add the operation to `operations.conf`:
+
+```HOCON
+{
+    operationId = get-books-gql
+    handlers = ${config.server.handlers.common.request} [
+        {
+            name = graphqlHandler
+            config = {
+                schema = "books.graphqls"
+            }
+        }
+    ]
+}
+```
+
+Note the `${config.server.handlers.common.request}` part. We import some standard configuration because we'll need it later. Now the only thing left are data fetchers and you'll have a working graphql endpoint.
+
+## Step 2
+We would like to fetch data from 3rd parties using existing Knot.x mechanisms, not to reinvent the wheel. Ideally we would have a generic data fetcher that can process Knot.x tasks defined in configuration. 
+
+Our fetcher will parse configuration to create tasks and run them using `FragmentsEngine`. It will output a `Fragment` that will contain data fetched from google in it's payload. Fragment's payload is just a `JsonObject` where actions can store data.
+
+
+#### Execute tasks from configuration
+Let's start with something like this:

```

Comment:
please provide some description of this `something` ;)

Reply:
This `something` is explained in details below the code snippet